### PR TITLE
chore(#8524): extract package.json subscripts

### PR DESCRIPTION
**Pull request by garethbowen**
# Description

[description]

medic/cht-core#8524

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs

If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:extract-subscripts/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:extract-subscripts/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:extract-subscripts/docker-compose/cht-couchdb-clustered.yml)

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by garethbowen** [comment]
@dianabarsan No hurry on this, but we discussed simplifying the package.json by moving all the subscripts into bash files. If you prefer this I'll raise an issue for it and merge.

I could go one step further and have each each DEV and CI script also just call a bash file. Let me know what you think!

**Review by dianabarsan — COMMENTED** [review]
Thanks so much for cleaning this up, this is much more readable, manageable and understandable now. I left a couple of nits inline.

File: scripts/build/build-angularjs-template-cache.js [hunk 1]
```diff
@@ -37,8 +38,19 @@ const getContent = async (file) => {
   );`;
 };
 
+const mkdirIfDoesNotExist = async (path) => {
+  try {
+    await mkdir(path, { recursive: true });
```
[the remark this answers is not in the document]
Comment (garethbowen) [reply scripts/build/build-angularjs-template-cache.js:43]: Nice catch! Fixed.

File: package.json [hunk 2]
```diff
@@ -18,78 +18,41 @@
...
     "build-dev-watch": "npm run build-dev && cd webapp && npm run build -- --configuration=development --watch=true & node ./scripts/build/watch.js",
-    "build-documentation": "npm run jsdoc-admin && npm run jsdoc-sentinel && npm run jsdoc-api && npm run jsdoc-shared-libs",
+    "build-documentation": "./scripts/build/build-documentation.sh",
     "build-webapp-dev": "cd webapp && npm run build -- --configuration=development && npm run compile",
-    "dev-api": "npm run copy-api-bowser && TZ=UTC nodemon --inspect=0.0.0.0:9229 --ignore 'api/build/static' --ignore 'api/build/public' --watch api --watch 'shared-libs/**/src/**' api/server.js -- --allow-cors",
+    "copy-api-resources": "cp -r api/src/public/* api/build/static/",
+    "default-wdio-mobile-local": "export VERSION=$(node ./scripts/build/get-version.js) && ./scripts/build/build-service-images.sh && wdio run ./tests/e2e/default-mobile/wdio.conf.js --suite=all",
+    "dev-api": "./scripts/build/copy-static-files.sh && TZ=UTC nodemon --inspect=0.0.0.0:9229 --ignore 'api/build/static' --ignore 'api/build/public' --watch api --watch 'shared-libs/**/src/**' api/server.js -- --allow-cors",
```
[the remark this answers is not in the document]
Comment (garethbowen) [reply package.json:27]: I've split the "dev scripts" into build and test groups. What do you think?

[the remark this answers is not in the document]
Comment (dianabarsan) [reply package.json:27]: Awesome, thanks!

**Review by dianabarsan — APPROVED** [review]
Much nicer, thank you!

**Pull request merged**